pkgs/node/src/chain.zig

```diff
@@ -402,10 +406,20 @@ pub const BeamChain = struct {
             block.slot,
         });
 
+        var missing_roots = std.ArrayList(types.Root).init(self.allocator);
+        errdefer missing_roots.deinit();
+
         const signatures = signedBlock.signature.constSlice();
+
         for (block.body.attestations.constSlice(), 0..) |attestation, index| {
             // Validate attestation before processing (from block = true)
             self.validateAttestation(attestation, true) catch |e| {
+                if (e == AttestationValidationError.UnknownHeadBlock) {
+                    missing_roots.append(attestation.data.head.root) catch |append_err| {
+                        self.module_logger.warn("Failed to track missing root: {any}", .{append_err});
```

Comment:
this error is unlkely, only in case of mem overflow, just use try and let the entire onblock fail, no need for excessive logging